tests/fixtures/transform_custom.py

```diff
@@ -29,3 +36,55 @@ def transform(X, transformer, y=None):
         )
     else:
         return pd.DataFrame(transformed)
+
+
+class CustomTask(TransformerInterface):
+    @classmethod
+    def load_task(cls, artifact_directory):
```

Comment:
can we solve this issue by having unique artifact directories instead of unique file names?

so something like

`tests/fixtures/drop_in_model_artifacts/sklearn_bin/<default_filename>`

Reply:
This default_filename seems wrong to me, or I just don't know/understand some logic behind this.

If adjustment is needed, I'd better copy `custom_artifact_name` to `default_filename` when constructing code_dir for a test.

Reply:
In normal usage of custom tasks users shouldn't need to worry about the filename for their pickled task, or even that it is pickled. Normally an empty output directory is used to dump the results of fit, and then that can be moved to wherever predict happens.  We also have a `save` and a `load` method for more compicated situations such as serializing keras or pytorch models.  The way we have these pickle files just saved and reused is an abnormal use case specific to how these tests were implemented.

Reply:
So I changed tests to rename artifacts to a provided name. For these test cases I provide drum_artifact.pkl